client/gutenberg/extensions/publicize/connection.jsx

```diff
@@ -81,25 +120,31 @@ class PublicizeConnection extends Component {
 			/>
 		);
 
-		if ( disabled || this.connectionIsFailing() ) {
+		if ( disabled || this.connectionIsFailing() || this.connectionNeedsReauth() ) {
 			toggle = <Disabled>{ toggle }</Disabled>;
 		}
 
 		return (
 			<li>
+				{ this.maybeDisplayGooglePlusNotice( serviceName ) }
+				{ this.maybeDisplayLinkedInNotice( serviceName ) }
 				<div className="publicize-jetpack-connection-container">
 					<label htmlFor={ fieldId } className="jetpack-publicize-connection-label">
 						<PublicizeServiceIcon serviceName={ serviceName } />
 						<span className="jetpack-publicize-connection-label-copy">{ label }</span>
 					</label>
 					{ toggle }
 				</div>
-				{ this.maybeDisplayGooglePlusNotice( serviceName ) }
 			</li>
 		);
 	}
 }
 
-export default withSelect( select => ( {
-	failedConnections: select( 'jetpack/publicize' ).getFailedConnections(),
-} ) )( PublicizeConnection );
+export default compose( [
+	withSelect( select => ( {
+		failedConnections: select( 'jetpack/publicize' ).getFailedConnections(),
+	} ) ),
+	withSelect( select => ( {
```

Comment:
We shouldn't need compose or multiple `withSelect`, can't we select all the values we need inside one call?

Reply:
Of course we can - it's also the same store, so we should connect once to it, and just pick the 2 state-derived props that are of interest for us.

Reply:
Consolidated in one.